java/src/main/native/ai_onnxruntime_OnnxTensor.c

```diff
@@ -213,20 +243,23 @@
  */
 JNIEXPORT jfloat JNICALL Java_ai_onnxruntime_OnnxTensor_getFloat
   (JNIEnv * jniEnv, jobject jobj, jlong apiHandle, jlong handle, jint onnxType) {
-    (void) jobj; // Required JNI parameter not needed by functions which don't need to access their host object.
-    const OrtApi* api = (const OrtApi*) apiHandle;
-    if (onnxType == 9) {
-        uint16_t* arr;
-        checkOrtStatus(jniEnv,api,api->GetTensorMutableData((OrtValue*)handle,(void**)&arr));
-        jfloat floatVal = convertHalfToFloat(*arr);
-        return floatVal;
-    } else if (onnxType == 10) {
-        jfloat* arr;
-        checkOrtStatus(jniEnv,api,api->GetTensorMutableData((OrtValue*)handle,(void**)&arr));
-        return *arr;
-    } else {
-        return NAN;
+  (void) jobj;  // Required JNI parameter not needed by functions which don't need to access their host object.
+  const OrtApi* api = (const OrtApi*) apiHandle;
+  if (onnxType == 9) {
+    uint16_t* arr = NULL;
+    OrtErrorCode code = checkOrtStatus(jniEnv, api, api->GetTensorMutableData((OrtValue*)handle, (void**)&arr));
+    if (code == ORT_OK) {
+      jfloat floatVal = convertHalfToFloat(*arr);
+      return floatVal;
+    }
+  } else if (onnxType == 10) {
+    jfloat* arr = NULL;
```

Comment:
> jfloat [](http://example.com/codeflow?start=3&length=7)

Is jfloat always 32 bits?

Reply:
Yes. All the java types are defined to be a fixed size across platforms.